{
  "url": "https://api.github.com/repos/facebook/react/issues/16199",
  "repository_url": "https://api.github.com/repos/facebook/react",
  "labels_url": "https://api.github.com/repos/facebook/react/issues/16199/labels{/name}",
  "comments_url": "https://api.github.com/repos/facebook/react/issues/16199/comments",
  "events_url": "https://api.github.com/repos/facebook/react/issues/16199/events",
  "html_url": "https://github.com/facebook/react/issues/16199",
  "id": 472643237,
  "node_id": "MDU6SXNzdWU0NzI2NDMyMzc=",
  "number": 16199,
  "title": "Error boundary reports \"fewer hooks\" unexpectedly",
  "user": {
    "login": "otakustay",
    "id": 639549,
    "node_id": "MDQ6VXNlcjYzOTU0OQ==",
    "avatar_url": "https://avatars2.githubusercontent.com/u/639549?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/otakustay",
    "html_url": "https://github.com/otakustay",
    "followers_url": "https://api.github.com/users/otakustay/followers",
    "following_url": "https://api.github.com/users/otakustay/following{/other_user}",
    "gists_url": "https://api.github.com/users/otakustay/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/otakustay/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/otakustay/subscriptions",
    "organizations_url": "https://api.github.com/users/otakustay/orgs",
    "repos_url": "https://api.github.com/users/otakustay/repos",
    "events_url": "https://api.github.com/users/otakustay/events{/privacy}",
    "received_events_url": "https://api.github.com/users/otakustay/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "closed",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 3,
  "created_at": "2019-07-25T03:13:34Z",
  "updated_at": "2019-07-27T12:40:11Z",
  "closed_at": "2019-07-27T12:40:11Z",
  "author_association": "NONE",
  "body": "<!--\r\n  Note: if the issue is about documentation or the website, please file it at:\r\n  https://github.com/reactjs/reactjs.org/issues/new\r\n-->\r\n\r\n**Do you want to request a *feature* or report a *bug*?**\r\n\r\nBUG\r\n\r\n**What is the current behavior?**\r\n\r\nWhen a error boundary receives error from its children and renders a custom error component, react reports \"Rendered fewer hooks than expected\" on the error component if we have some hooks after the `throw` statement\r\n\r\n**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have dependencies other than React. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**\r\n\r\nhttps://codesandbox.io/s/error-boundary-hooks-mismatch-tj52l\r\n\r\nAfter clicks \"INC\" button several times, the page will crash instead of rendering an error label:\r\n\r\n```\r\nUncaught Invariant Violation: Rendered fewer hooks than expected. This may be caused by an accidental early return statement.\r\n    at invariant (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:55:15)\r\n    at renderWithHooks (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:13010:28)\r\n    at mountIndeterminateComponent (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:15020:13)\r\n    at beginWork (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:15625:16)\r\n    at performUnitOfWork (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:19312:12)\r\n    at workLoop (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:19352:24)\r\n    at renderRoot (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:19435:7)\r\n    at performWorkOnRoot (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:20342:7)\r\n    at performWork (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:20254:7)\r\n    at performSyncWork (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:20228:3)\r\n    at interactiveUpdates$1 (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:20495:7)\r\n    at interactiveUpdates (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:2170:10)\r\n    at dispatchInteractiveEvent (https://tj52l.csb.app/node_modules/react-dom/cjs/react-dom.development.js:4882:3)\r\ninvariant @ react-dom.development.js:55\r\nrenderWithHooks @ react-dom.development.js:13010\r\nmountIndeterminateComponent @ react-dom.development.js:15020\r\nbeginWork @ react-dom.development.js:15625\r\nperformUnitOfWork @ react-dom.development.js:19312\r\nworkLoop @ react-dom.development.js:19352\r\nrenderRoot @ react-dom.development.js:19435\r\nperformWorkOnRoot @ react-dom.development.js:20342\r\nperformWork @ react-dom.development.js:20254\r\nperformSyncWork @ react-dom.development.js:20228\r\ninteractiveUpdates$1 @ react-dom.development.js:20495\r\ninteractiveUpdates @ react-dom.development.js:2170\r\ndispatchInteractiveEvent @ react-dom.development.js:4882\r\n\r\nThe above error occurred in the <ErrorLabel> component:\r\n    in ErrorLabel (created by _default)\r\n    in div (created by _default)\r\n    in _default\r\n```\r\n\r\n**What is the expected behavior?**\r\n\r\nAlthough there are some hooks after the `throw` statement in `App` component, it should not report hooks mismatch since the render of `App` is aborted and component type is switched from `App` to `ErrorLabel`.\r\n\r\nSecond, it should not report error at `ErrorLabel` component, `ErrorLabel` is only a simple presentational component without any hooks.\r\n\r\n**Which versions of React, and which browser / OS are affected by this issue? Did this work in previous versions of React?**\r\n\r\n- react: 16.8.6\r\n- react-dom: 16.8.6",
  "closed_by": {
    "login": "threepointone",
    "id": 18808,
    "node_id": "MDQ6VXNlcjE4ODA4",
    "avatar_url": "https://avatars2.githubusercontent.com/u/18808?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/threepointone",
    "html_url": "https://github.com/threepointone",
    "followers_url": "https://api.github.com/users/threepointone/followers",
    "following_url": "https://api.github.com/users/threepointone/following{/other_user}",
    "gists_url": "https://api.github.com/users/threepointone/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/threepointone/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/threepointone/subscriptions",
    "organizations_url": "https://api.github.com/users/threepointone/orgs",
    "repos_url": "https://api.github.com/users/threepointone/repos",
    "events_url": "https://api.github.com/users/threepointone/events{/privacy}",
    "received_events_url": "https://api.github.com/users/threepointone/received_events",
    "type": "User",
    "site_admin": false
  }
}
